### DISCUSSION TITLE
How to LogisticsServiceCharges

### DISCUSSION THREAD

**Discussion by GeorgDczg** [stephanstapel/ZUGFeRD-csharp#284] [Q&A] [answered]
Dear all,

I wonder how I can specify the shipping costs in an invoice? 
I am able to add the costs like in the main description:
desc.AddLogisticsServiceCharge(5.80m, "Versandkosten", TaxTypes.VAT, TaxCategoryCodes.S, 7m);

Unfortunately, the values are not saved using
desc.Save(strFile, s2industries.ZUGFeRD.ZUGFeRDVersion.Version21, s2industries.ZUGFeRD.Profile.XRechnung1);
desc.Save(strFile, s2industries.ZUGFeRD.ZUGFeRDVersion.Version21, s2industries.ZUGFeRD.Profile.XRechnung);

Could you point me in the right direction? 

Thanks in advance!
Georg

**Comment by stephanstapel** [comment, marked as answer]
Hi @GeorgDczg ,

Thanks for your question. Both standards, ZUGFeRD and XRechnung have developed quickly and the same is true for the library. Unfortunately, ram:SpecifiedLogisticsServiceCharge is not part of the XRechnung specification. I have added a documentation to the code now to clarify this.

**Comment by GeorgDczg** [comment]
Hallo Stephan, thank you for your reply!
I set the service charges for totals and SpecifiedLogisticsServiceCharge wich are not exportet so I get a
[BR-CO-12]-Sum of charges on document level (BT-108) = Σ Document level charge amount (BT-99).

What is the solution from your point of view? Is AddTradeAllowanceCharge perhaps the right way to go?

**Comment by GeorgDczg** [comment]
AddTradeAllowanceCharge has worked fine for me.
